File: DesignPatternsExercises/java/PrototypeDesignPattern/Solution/PrototypeDesignPattern.java
public class PrototypeDesignPattern {
    public static void main(String[] args) {
        try {
            Car originalCar = new Car("Toyota");
            // Car clonedCar  = __________(1)__________ PrototypeFactory.getInstance(originalCar);
            Car clonedCar = (Car) PrototypeFactory.getInstance(originalCar);
            clonedCar.setModel("Honda");

            System.out.println("\nOriginal Car:");
            originalCar.printDetails();
            System.out.println("\nCloned Car:");
            clonedCar.printDetails();
        } catch (CloneNotSupportedException e) {
            System.out.println("Object not cloneable");
            e.printStackTrace();
        }
    }
}

File: DesignPatternsExercises/ruby/prototype_design_pattern/exercise/prototype_design_pattern.rb
require_relative 'car'
require_relative 'prototype_factory'

original_car = Car.new(:Toyota)
cloned_car = PrototypeFactory.instance(original_car)
cloned_car.model = :Honda

puts "\nOriginal Car:"
original_car.print_details
puts "\nCloned Car:"
cloned_car.print_details
